app/actions/views/channel.js

```diff
@@ -273,6 +274,38 @@ export function selectInitialChannel(teamId) {
     };
 }
 
+export function selectPenultimateChannel(teamId) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const {channels, myMembers} = state.entities.channels;
+        const {currentUserId} = state.entities.users;
+        const {myPreferences} = state.entities.preferences;
+        const viewArchivedChannels = getConfig(state).ExperimentalViewArchivedChannels === 'true';
+        const lastChannelForTeam = state.views.team.lastChannelForTeam[teamId];
+        const lastChannelId = lastChannelForTeam && lastChannelForTeam.length > 1 ? lastChannelForTeam[1] : '';
+        const lastChannel = channels[lastChannelId];
+
+        const isDMVisible = lastChannel && lastChannel.type === General.DM_CHANNEL &&
+            isDirectChannelVisible(currentUserId, myPreferences, lastChannel);
+
+        const isGMVisible = lastChannel && lastChannel.type === General.GM_CHANNEL &&
+            isGroupChannelVisible(myPreferences, lastChannel);
+
+        if (
+            myMembers[lastChannelId] &&
+            lastChannel &&
+            (lastChannel.delete_at === 0 || viewArchivedChannels) &&
+            (lastChannel.team_id === teamId || isDMVisible || isGMVisible)
+        ) {
+            handleSelectChannel(lastChannelId)(dispatch, getState);
```

Comment:
change it to `dispatch(handleSelectChannel(lastChannelId))` same thing with the other one